pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -264,6 +263,21 @@ func (r *Reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, addo
 		return reqeueAfter, nil
 	}
 
+	// This handles manifests that should be removed manually, because kubectl --prune does not always delete obsolete resources.
+	if cluster.Spec.Cloud.VSphere != nil && addon.Name == "csi" {
+		legacyAddon := &kubermaticv1.Addon{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "internal-delete",
+			},
+			Spec: kubermaticv1.AddonSpec{
+				Name: "internal-delete",
+			},
+		}
+		if err := r.cleanupManifests(ctx, log, legacyAddon, cluster); err != nil {
```

Comment:
Is this going to error if it's already deleted or wasn't deployed in the first place?

Reply:
It uses kubectl under the hood with `--ignore-not-found` as parameter.